File: DHealth/app/src/main/java/com/example/hamid/dhealth/di/module/PreferenceModule.java
package com.example.hamid.dhealth.di.module;

import com.example.hamid.dhealth.data.Preference.PreferenceManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PreferenceModule {

    @Provides
    @Singleton
    PreferenceManager providePreferenceManager() {
        return new PreferenceManager();
    }
}

File: DHealth-Android/Kotlin/app/src/main/java/com/example/hamid/dhealth/di/module/PreferenceModule.kt
package com.example.hamid.dhealth.di.module

import com.example.hamid.dhealth.data.Preference.PreferenceManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PreferenceModule {

    @Provides
    @Singleton
    fun providePreferenceManager(): PreferenceManager {
        return PreferenceManager()
    }
}
